internal/controller/datadogagent/feature/clusterchecks/feature.go

```diff
@@ -72,9 +72,18 @@ func (f *clusterChecksFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp fe
 
 		f.useClusterCheckRunners = apiutils.BoolValue(dda.Spec.Features.ClusterChecks.UseClusterChecksRunners)
 		reqComp = feature.RequiredComponents{
-			Agent:               feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
-			ClusterAgent:        feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
-			ClusterChecksRunner: feature.RequiredComponent{IsRequired: &f.useClusterCheckRunners},
+			Agent: feature.RequiredComponent{
+				IsRequired: apiutils.NewBoolPointer(true),
+				Containers: []apicommon.AgentContainerName{apicommon.CoreAgentContainerName},
```

Comment:
That's out of scope for this PR, but technically, we don't need core Agent if useclustercheckrunners is enabled

Reply:
Agree. That'll be good for a future PR